In [ ]:
%pip install panda
import pandas as pd

In [145]:
pd.set_option('display.max_columns', None)

In [146]:
arquivos = {
    '2020': './esus-vepi.LeitoOcupacao_2020.csv',
    '2021': './esus-vepi.LeitoOcupacao_2021.csv',
    '2022': './esus-vepi.LeitoOcupacao_2022.csv'
}

lista_dataframes = []

for ano, arquivo in arquivos.items():
    try:
        df = pd.read_csv(arquivo, encoding='utf-8')

        df['ano_origem'] = ano

        lista_dataframes.append(df)
    except Exception as e:
        print(f"Erro ao ler os dados do ano {ano}: {e}")


df = pd.concat(lista_dataframes, ignore_index=True) #Concatenando todos os DFs
print(f"Total de linhas no df unificado: {len(df)}")

C:\Users\Davi2\AppData\Local\Temp\ipykernel_16472\4088843205.py:11: DtypeWarning: Columns (0: cnes) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(arquivo, encoding='utf-8')


Total de linhas no df unificado: 1591675


# Análise do dataset

In [147]:
display(df.head(5))
display(df.info())

,Unnamed: 0,_id,dataNotificacao,cnes,ocupacaoSuspeitoCli,ocupacaoSuspeitoUti,ocupacaoConfirmadoCli,ocupacaoConfirmadoUti,ocupacaoCovidUti,ocupacaoCovidCli,ocupacaoHospitalarUti,ocupacaoHospitalarCli,saidaSuspeitaObitos,saidaSuspeitaAltas,saidaConfirmadaObitos,saidaConfirmadaAltas,origem,_p_usuario,estadoNotificacao,municipioNotificacao,estado,municipio,excluido,validado,_created_at,_updated_at,ano_origem
0,0,p5Ez41Zu6B,2020-04-14T03:00:00.000Z,2303167,0.0,0.0,0.0,0.0,NaN,NaN,NaN,NaN,0.0,0.0,0.0,0.0,aplicacao-web,_User$auVMjEVEVz,Santa Catarina,Itapema,Santa Catarina,Itapema,True,False,2020-04-15T13:44:30.085Z,2020-04-16T13:56:13.656Z,2020
1,1,eUOMsSt7T7,2020-04-15T03:00:00.000Z,5935377\t,1.0,0.0,0.0,0.0,NaN,NaN,NaN,NaN,0.0,1.0,0.0,0.0,parse-cloud,_User$WBa5nfe9P9,Rio de Janeiro,Rio de Janeiro,Rio de Janeiro,Rio de Janeiro,False,False,2020-04-15T13:50:47.256Z,2020-08-01T20:08:13.116Z,2020
2,2,QaG4oIBrMa,2020-04-15T14:57:16.255Z,0012599,15.0,1.0,3.0,2.0,NaN,NaN,NaN,NaN,0.0,0.0,0.0,0.0,parse-cloud,_User$6oxZQHBPQ7,Rio de Janeiro,Niterói,Rio de Janeiro,Niterói,False,False,2020-04-15T15:20:57.033Z,2020-08-01T20:02:12.729Z,2020
3,3,ifya31F3ZF,2020-03-24T03:00:00.000Z,2562871,9.0,0.0,1.0,0.0,NaN,NaN,NaN,NaN,0.0,0.0,0.0,0.0,parse-cloud,_User$fhufIbmR9C,Ceará,Várzea Alegre,Ceará,Várzea Alegre,False,False,2020-04-15T16:58:32.839Z,2020-08-01T21:40:37.624Z,2020
4,4,dkurDhelx9,2020-04-01T03:00:00.000Z,2252228,26.0,0.0,1.0,0.0,NaN,NaN,NaN,NaN,0.0,0.0,0.0,0.0,parse-cloud,_User$tkAdHTNhC9,Rio Grande do Sul,Encantado,Rio Grande do Sul,Encantado,False,False,2020-04-15T17:02:53.877Z,2020-08-01T21:37:53.664Z,2020


<class 'pandas.DataFrame'>
RangeIndex: 1591675 entries, 0 to 1591674
Data columns (total 27 columns):
 #   Column                 Non-Null Count    Dtype  
---  ------                 --------------    -----  
 0   Unnamed: 0             1591675 non-null  int64  
 1   _id                    1591675 non-null  str    
 2   dataNotificacao        1591675 non-null  str    
 3   cnes                   1591625 non-null  object 
 4   ocupacaoSuspeitoCli    1014964 non-null  float64
 5   ocupacaoSuspeitoUti    1014964 non-null  float64
 6   ocupacaoConfirmadoCli  1014965 non-null  float64
 7   ocupacaoConfirmadoUti  1014965 non-null  float64
 8   ocupacaoCovidUti       583542 non-null   float64
 9   ocupacaoCovidCli       583542 non-null   float64
 10  ocupacaoHospitalarUti  583542 non-null   float64
 11  ocupacaoHospitalarCli  583542 non-null   float64
 12  saidaSuspeitaObitos    1591559 non-null  float64
 13  saidaSuspeitaAltas     1591571 non-null  float64
 14  saidaConfirmadaObitos  159166

None

In [148]:
df.nunique(dropna=False)

Unnamed: 0               1591675
_id                      1591675
dataNotificacao           799916
cnes                        7635
ocupacaoSuspeitoCli         1129
ocupacaoSuspeitoUti          554
ocupacaoConfirmadoCli        596
ocupacaoConfirmadoUti        506
ocupacaoCovidUti             225
ocupacaoCovidCli             227
ocupacaoHospitalarUti        330
ocupacaoHospitalarCli        660
saidaSuspeitaObitos          196
saidaSuspeitaAltas           398
saidaConfirmadaObitos        393
saidaConfirmadaAltas        1234
origem                        13
_p_usuario                  3835
estadoNotificacao             28
municipioNotificacao        2353
estado                        29
municipio                   2355
excluido                       2
validado                       1
_created_at              1590872
_updated_at              1590902
ano_origem                     3
dtype: int64

# cnes não inteiro

In [149]:
cnes_num = pd.to_numeric(df['cnes'], errors='coerce')

print("Invalid CNES rows:", cnes_num.isna().sum())
print("Valid CNES rows:", cnes_num.notna().sum())

Invalid CNES rows: 1672
Valid CNES rows: 1590003


In [150]:
cnes_num = pd.to_numeric(df['cnes'], errors='coerce')

df = df[cnes_num.notna()].copy()

df['cnes'] = cnes_num[cnes_num.notna()].astype('int64')

In [151]:
cnes_num = pd.to_numeric(df['cnes'], errors='coerce')

print("Invalid CNES rows:", cnes_num.isna().sum())
print("Valid CNES rows:", cnes_num.notna().sum())

Invalid CNES rows: 0
Valid CNES rows: 1590003


# Valores únicos

Investigando colunas com quantidade razoável de valores únicos

In [152]:
cols = ['origem', 'estado', 'excluido', 'validado']

for col in cols:
    print(f"\n=== {col} ===")
    print(df[col].unique())
    print(f"Quantidade de valores únicos: {df[col].nunique(dropna=False)}")


=== origem ===
<StringArray>
[        'aplicacao-web',           'parse-cloud',                'RPA-SP',
                'RPA-CE',                'RPA-RS',                'RPA-MS',
                'RPA-GO',                'RPA-MT',                'RPA-RN',
                'RPA-PR',                'RPA-MG', 'RPA-MG-BELO-HORIZONTE',
       'RPA-PR-CURITIBA']
Length: 13, dtype: str
Quantidade de valores únicos: 13

=== estado ===
<StringArray>
[     'Santa Catarina',      'Rio de Janeiro',               'Ceará',
   'Rio Grande do Sul',        'Minas Gerais',                'Pará',
               'Piauí',  'Mato Grosso do Sul',               'Goiás',
          'Pernambuco',            'Rondônia',                'Acre',
            'Amazonas',              'Paraná',            'Maranhão',
               'Bahia',           'São Paulo',             'Paraíba',
             'Alagoas',         'Mato Grosso',             'Sergipe',
    'Distrito Federal',           'Tocantins', 'Rio Grande do No

Observe que a coluna validado tem valor False para todas as entradas logo ela não tem valor e será removida do dataset final

In [153]:
df = df.drop(columns=['validado'])

Os valores RPA-PR-CURITIBA e RPA-MG-BELO-HORIZONTE serão agrupados em RPA-PR e RPA-MG respectivamente

In [154]:
df['origem'] = df['origem'].replace('RPA-PR-CURITIBA', 'RPA-PR')
df['origem'] = df['origem'].replace('RPA-MG-BELO-HORIZONTE', 'RPA-MG')

Ajustar GOIAS para Goiás para ficar padronizado

In [155]:
df['estado'] = df['estado'].replace('GOIAS', 'Goiás')

In [156]:
cols = ['origem', 'estado', 'excluido']

for col in cols:
    print(f"\n=== {col} ===")
    print(df[col].unique())
    print(f"Quantidade de valores únicos: {df[col].nunique(dropna=False)}")


=== origem ===
<StringArray>
['aplicacao-web',   'parse-cloud',        'RPA-SP',        'RPA-CE',
        'RPA-RS',        'RPA-MS',        'RPA-GO',        'RPA-MT',
        'RPA-RN',        'RPA-PR',        'RPA-MG']
Length: 11, dtype: str
Quantidade de valores únicos: 11

=== estado ===
<StringArray>
[     'Santa Catarina',      'Rio de Janeiro',               'Ceará',
   'Rio Grande do Sul',        'Minas Gerais',                'Pará',
               'Piauí',  'Mato Grosso do Sul',               'Goiás',
          'Pernambuco',            'Rondônia',                'Acre',
            'Amazonas',              'Paraná',            'Maranhão',
               'Bahia',           'São Paulo',             'Paraíba',
             'Alagoas',         'Mato Grosso',             'Sergipe',
    'Distrito Federal',           'Tocantins', 'Rio Grande do Norte',
                   nan,               'Amapá',      'Espírito Santo',
             'Roraima']
Length: 28, dtype: str
Quantidade de val

# Tratando NaN

In [157]:
print("Valores ausentes por coluna:")
print(df.isnull().sum())

Valores ausentes por coluna:
Unnamed: 0                     0
_id                            0
dataNotificacao                0
cnes                           0
ocupacaoSuspeitoCli       575215
ocupacaoSuspeitoUti       575215
ocupacaoConfirmadoCli     575214
ocupacaoConfirmadoUti     575214
ocupacaoCovidUti         1007957
ocupacaoCovidCli         1007957
ocupacaoHospitalarUti    1007957
ocupacaoHospitalarCli    1007957
saidaSuspeitaObitos          116
saidaSuspeitaAltas           104
saidaConfirmadaObitos         12
saidaConfirmadaAltas          13
origem                         0
_p_usuario                     0
estadoNotificacao              0
municipioNotificacao           0
estado                         1
municipio                      1
excluido                       0
_created_at                    0
_updated_at                    0
ano_origem                     0
dtype: int64


## cnes nulo

Colunas com cnes nulo serão removidas já que não tem como inferir essa informação.

In [158]:
df['cnes'].isna().sum()

np.int64(0)

In [159]:
df = df.dropna(subset=['cnes'])

In [160]:
df['cnes'].isna().sum()

np.int64(0)

In [161]:
print("Valores ausentes por coluna:")
print(df.isnull().sum())

Valores ausentes por coluna:
Unnamed: 0                     0
_id                            0
dataNotificacao                0
cnes                           0
ocupacaoSuspeitoCli       575215
ocupacaoSuspeitoUti       575215
ocupacaoConfirmadoCli     575214
ocupacaoConfirmadoUti     575214
ocupacaoCovidUti         1007957
ocupacaoCovidCli         1007957
ocupacaoHospitalarUti    1007957
ocupacaoHospitalarCli    1007957
saidaSuspeitaObitos          116
saidaSuspeitaAltas           104
saidaConfirmadaObitos         12
saidaConfirmadaAltas          13
origem                         0
_p_usuario                     0
estadoNotificacao              0
municipioNotificacao           0
estado                         1
municipio                      1
excluido                       0
_created_at                    0
_updated_at                    0
ano_origem                     0
dtype: int64


## Tratamento de NaN para a coluna 'estado'

In [162]:
df[df['estado'].isna()].head()

,Unnamed: 0,_id,dataNotificacao,cnes,ocupacaoSuspeitoCli,ocupacaoSuspeitoUti,ocupacaoConfirmadoCli,ocupacaoConfirmadoUti,ocupacaoCovidUti,ocupacaoCovidCli,ocupacaoHospitalarUti,ocupacaoHospitalarCli,saidaSuspeitaObitos,saidaSuspeitaAltas,saidaConfirmadaObitos,saidaConfirmadaAltas,origem,_p_usuario,estadoNotificacao,municipioNotificacao,estado,municipio,excluido,_created_at,_updated_at,ano_origem
4182,4184,bc66V3say8,2020-04-24T12:32:29.807Z,2246902,1.0,0.0,1.0,0.0,NaN,NaN,NaN,NaN,0.0,1.0,0.0,1.0,aplicacao-web,_User$Ow1jWbSgxK,Rio Grande do Sul,Não-Me-Toque,NaN,NaN,False,2020-04-24T12:33:53.952Z,2020-04-24T12:33:53.952Z,2020


Olhando outras entradas com cnes "2246902" verifica-se que o estado é Rio Grande do Sul e o municipio é Não-Me-Toque.

In [163]:
df[df["cnes"] == 2246902].head()

,Unnamed: 0,_id,dataNotificacao,cnes,ocupacaoSuspeitoCli,ocupacaoSuspeitoUti,ocupacaoConfirmadoCli,ocupacaoConfirmadoUti,ocupacaoCovidUti,ocupacaoCovidCli,ocupacaoHospitalarUti,ocupacaoHospitalarCli,saidaSuspeitaObitos,saidaSuspeitaAltas,saidaConfirmadaObitos,saidaConfirmadaAltas,origem,_p_usuario,estadoNotificacao,municipioNotificacao,estado,municipio,excluido,_created_at,_updated_at,ano_origem
3905,3907,5iesHj3Fzc,2020-04-23T16:46:06.402Z,2246902,5.0,0.0,0.0,0.0,NaN,NaN,NaN,NaN,0.0,0.0,0.0,0.0,parse-cloud,_User$Ow1jWbSgxK,Rio Grande do Sul,Não-Me-Toque,Rio Grande do Sul,Não-Me-Toque,False,2020-04-23T16:47:02.772Z,2020-08-01T18:55:29.006Z,2020
4182,4184,bc66V3say8,2020-04-24T12:32:29.807Z,2246902,1.0,0.0,1.0,0.0,NaN,NaN,NaN,NaN,0.0,1.0,0.0,1.0,aplicacao-web,_User$Ow1jWbSgxK,Rio Grande do Sul,Não-Me-Toque,NaN,NaN,False,2020-04-24T12:33:53.952Z,2020-04-24T12:33:53.952Z,2020
4185,4187,T5m9LQ3nOz,2020-04-14T03:00:00.000Z,2246902,1.0,0.0,1.0,0.0,NaN,NaN,NaN,NaN,0.0,1.0,0.0,1.0,parse-cloud,_User$Ow1jWbSgxK,Rio Grande do Sul,Não-Me-Toque,Rio Grande do Sul,Não-Me-Toque,False,2020-04-24T12:37:16.319Z,2020-08-01T20:12:50.278Z,2020
4186,4188,MrFGgYzjdJ,2020-04-16T03:00:00.000Z,2246902,1.0,0.0,1.0,0.0,NaN,NaN,NaN,NaN,0.0,1.0,0.0,1.0,parse-cloud,_User$Ow1jWbSgxK,Rio Grande do Sul,Não-Me-Toque,Rio Grande do Sul,Não-Me-Toque,False,2020-04-24T12:37:54.071Z,2020-08-01T19:59:36.523Z,2020
4187,4189,CeWuzFylR1,2020-04-21T03:00:00.000Z,2246902,1.0,0.0,1.0,0.0,NaN,NaN,NaN,NaN,0.0,1.0,0.0,1.0,parse-cloud,_User$Ow1jWbSgxK,Rio Grande do Sul,Não-Me-Toque,Rio Grande do Sul,Não-Me-Toque,False,2020-04-24T12:38:22.763Z,2020-08-01T19:19:18.653Z,2020


In [164]:
df.loc[
    df['cnes'] == '2246902',
    ['estado', 'municipio']
] = ['Rio Grande do Sul', 'Não-Me-Toque']

In [165]:
df[df['estado'].isna()].head()

,Unnamed: 0,_id,dataNotificacao,cnes,ocupacaoSuspeitoCli,ocupacaoSuspeitoUti,ocupacaoConfirmadoCli,ocupacaoConfirmadoUti,ocupacaoCovidUti,ocupacaoCovidCli,ocupacaoHospitalarUti,ocupacaoHospitalarCli,saidaSuspeitaObitos,saidaSuspeitaAltas,saidaConfirmadaObitos,saidaConfirmadaAltas,origem,_p_usuario,estadoNotificacao,municipioNotificacao,estado,municipio,excluido,_created_at,_updated_at,ano_origem
4182,4184,bc66V3say8,2020-04-24T12:32:29.807Z,2246902,1.0,0.0,1.0,0.0,NaN,NaN,NaN,NaN,0.0,1.0,0.0,1.0,aplicacao-web,_User$Ow1jWbSgxK,Rio Grande do Sul,Não-Me-Toque,NaN,NaN,False,2020-04-24T12:33:53.952Z,2020-04-24T12:33:53.952Z,2020


## saída

df[df['saidaConfirmadaObitos'].isna()].head()

In [166]:
df[df['saidaConfirmadaObitos'].isna()].head(50)

,Unnamed: 0,_id,dataNotificacao,cnes,ocupacaoSuspeitoCli,ocupacaoSuspeitoUti,ocupacaoConfirmadoCli,ocupacaoConfirmadoUti,ocupacaoCovidUti,ocupacaoCovidCli,ocupacaoHospitalarUti,ocupacaoHospitalarCli,saidaSuspeitaObitos,saidaSuspeitaAltas,saidaConfirmadaObitos,saidaConfirmadaAltas,origem,_p_usuario,estadoNotificacao,municipioNotificacao,estado,municipio,excluido,_created_at,_updated_at,ano_origem
29249,29260,Pb3Z5MezCQ,2020-05-18T14:02:41.949Z,2012766,15.0,1.0,1.0,0.0,NaN,NaN,NaN,NaN,0.0,0.0,NaN,NaN,parse-cloud,_User$kESz5jJHJr,Amazonas,Manaus,Amazonas,Manaus,False,2020-05-18T14:04:20.066Z,2020-08-01T12:03:39.961Z,2020
29257,29268,vwOdDwIsyX,2020-05-18T14:02:41.949Z,2012766,15.0,1.0,1.0,0.0,NaN,NaN,NaN,NaN,0.0,0.0,NaN,NaN,parse-cloud,_User$kESz5jJHJr,Amazonas,Manaus,Amazonas,Manaus,False,2020-05-18T14:05:32.277Z,2020-08-01T12:03:38.649Z,2020
29258,29269,rvG7MhMwhD,2020-05-18T14:02:41.949Z,2012766,15.0,1.0,1.0,0.0,NaN,NaN,NaN,NaN,0.0,0.0,NaN,NaN,parse-cloud,_User$kESz5jJHJr,Amazonas,Manaus,Amazonas,Manaus,False,2020-05-18T14:07:58.038Z,2020-08-01T12:03:37.839Z,2020
29259,29270,NAHmolhvf6,2020-05-18T14:02:41.949Z,2012766,15.0,1.0,1.0,0.0,NaN,NaN,NaN,NaN,0.0,0.0,NaN,NaN,parse-cloud,_User$kESz5jJHJr,Amazonas,Manaus,Amazonas,Manaus,False,2020-05-18T14:07:51.148Z,2020-08-01T12:03:36.510Z,2020
29272,29283,DCQJOQoyWU,2020-05-18T14:02:41.949Z,2012766,15.0,1.0,1.0,0.0,NaN,NaN,NaN,NaN,0.0,0.0,NaN,NaN,parse-cloud,_User$kESz5jJHJr,Amazonas,Manaus,Amazonas,Manaus,False,2020-05-18T14:07:36.792Z,2020-08-01T12:03:34.852Z,2020
45567,45586,bZKmcbG4Ql,2020-05-28T03:00:00.000Z,2081253,35.0,12.0,10.0,3.0,NaN,NaN,NaN,NaN,0.0,0.0,NaN,NaN,parse-cloud,_User$Wc8lJNxzUO,São Paulo,Araras,São Paulo,Araras,False,2020-05-29T11:56:24.669Z,2020-08-01T08:07:05.736Z,2020
140248,140281,pLsyCdJYaS,2020-07-11T03:00:00.000Z,2561395,42.0,0.0,2.0,0.0,NaN,NaN,NaN,NaN,0.0,1.0,NaN,NaN,parse-cloud,_User$bkiCmE6DHy,Ceará,Aquiraz,Ceará,Aquiraz,False,2020-07-20T17:52:57.213Z,2020-07-31T18:42:48.564Z,2020
140252,140285,baBV97tRls,2020-07-12T03:00:00.000Z,2561395,42.0,0.0,2.0,0.0,NaN,NaN,NaN,NaN,0.0,0.0,NaN,NaN,parse-cloud,_User$bkiCmE6DHy,Ceará,Aquiraz,Ceará,Aquiraz,False,2020-07-20T17:53:22.299Z,2020-07-31T18:22:33.675Z,2020
140258,140291,pZ1kZD8neJ,2020-07-13T03:00:00.000Z,2561395,42.0,0.0,2.0,0.0,NaN,NaN,NaN,NaN,0.0,0.0,NaN,NaN,parse-cloud,_User$bkiCmE6DHy,Ceará,Aquiraz,Ceará,Aquiraz,False,2020-07-20T17:53:37.366Z,2020-07-31T18:02:19.879Z,2020
140260,140293,0TAzVjuIHr,2020-07-14T03:00:00.000Z,2561395,42.0,0.0,1.0,0.0,NaN,NaN,NaN,NaN,1.0,0.0,NaN,NaN,parse-cloud,_User$bkiCmE6DHy,Ceará,Aquiraz,Ceará,Aquiraz,False,2020-07-20T17:53:55.205Z,2020-07-31T17:39:47.756Z,2020


In [167]:
df[df["cnes"] == 2012766].head(50)

,Unnamed: 0,_id,dataNotificacao,cnes,ocupacaoSuspeitoCli,ocupacaoSuspeitoUti,ocupacaoConfirmadoCli,ocupacaoConfirmadoUti,ocupacaoCovidUti,ocupacaoCovidCli,ocupacaoHospitalarUti,ocupacaoHospitalarCli,saidaSuspeitaObitos,saidaSuspeitaAltas,saidaConfirmadaObitos,saidaConfirmadaAltas,origem,_p_usuario,estadoNotificacao,municipioNotificacao,estado,municipio,excluido,_created_at,_updated_at,ano_origem
14379,14383,T6uyj8YTzd,2020-05-05T16:23:42.983Z,2012766,15.0,1.0,8.0,0.0,NaN,NaN,NaN,NaN,1.0,0.0,1.0,2.0,parse-cloud,_User$0Ik365If7Z,Amazonas,Manaus,Amazonas,Manaus,False,2020-05-05T16:31:44.431Z,2020-08-01T16:26:45.559Z,2020
15098,15102,SK9jgoXpjr,2020-05-06T13:51:37.914Z,2012766,15.0,1.0,6.0,0.0,NaN,NaN,NaN,NaN,0.0,0.0,1.0,2.0,parse-cloud,_User$kESz5jJHJr,Amazonas,Manaus,Amazonas,Manaus,False,2020-05-06T13:53:52.540Z,2020-08-01T16:14:07.844Z,2020
17494,17498,xMxDujkHzz,2020-05-08T16:59:55.064Z,2012766,15.0,1.0,13.0,0.0,NaN,NaN,NaN,NaN,0.0,0.0,1.0,0.0,parse-cloud,_User$0Ik365If7Z,Amazonas,Manaus,Amazonas,Manaus,False,2020-05-08T17:00:47.427Z,2020-08-01T15:37:57.486Z,2020
21222,21231,pG5iPS1RiP,2020-05-12T14:00:06.715Z,2012766,15.0,1.0,3.0,0.0,NaN,NaN,NaN,NaN,1.0,0.0,0.0,0.0,parse-cloud,_User$kESz5jJHJr,Amazonas,Manaus,Amazonas,Manaus,False,2020-05-12T14:01:02.063Z,2020-08-01T14:19:06.868Z,2020
22853,22862,ta7HYBOfm7,2020-05-13T14:09:42.419Z,2012766,15.0,1.0,3.0,0.0,NaN,NaN,NaN,NaN,0.0,3.0,0.0,1.0,parse-cloud,_User$0Ik365If7Z,Amazonas,Manaus,Amazonas,Manaus,False,2020-05-13T14:10:34.198Z,2020-08-01T13:56:33.985Z,2020
24459,24468,6HiAKIwZef,2020-05-14T14:43:03.417Z,2012766,15.0,1.0,1.0,0.0,NaN,NaN,NaN,NaN,1.0,0.0,0.0,0.0,parse-cloud,_User$0Ik365If7Z,Amazonas,Manaus,Amazonas,Manaus,False,2020-05-14T15:03:48.337Z,2020-08-01T13:33:09.283Z,2020
25783,25794,Vy1J9DwBkC,2020-05-15T13:28:14.278Z,2012766,15.0,1.0,1.0,1.0,NaN,NaN,NaN,NaN,0.0,0.0,0.0,0.0,parse-cloud,_User$kESz5jJHJr,Amazonas,Manaus,Amazonas,Manaus,False,2020-05-15T13:30:31.962Z,2020-08-01T13:11:27.110Z,2020
27888,27899,bdAYe0SVu7,2020-05-17T13:09:47.444Z,2012766,15.0,0.0,0.0,0.0,NaN,NaN,NaN,NaN,0.0,0.0,0.0,0.0,parse-cloud,_User$0Ik365If7Z,Amazonas,Manaus,Amazonas,Manaus,False,2020-05-17T13:10:18.213Z,2020-08-01T12:27:22.077Z,2020
29249,29260,Pb3Z5MezCQ,2020-05-18T14:02:41.949Z,2012766,15.0,1.0,1.0,0.0,NaN,NaN,NaN,NaN,0.0,0.0,NaN,NaN,parse-cloud,_User$kESz5jJHJr,Amazonas,Manaus,Amazonas,Manaus,False,2020-05-18T14:04:20.066Z,2020-08-01T12:03:39.961Z,2020
29257,29268,vwOdDwIsyX,2020-05-18T14:02:41.949Z,2012766,15.0,1.0,1.0,0.0,NaN,NaN,NaN,NaN,0.0,0.0,NaN,NaN,parse-cloud,_User$kESz5jJHJr,Amazonas,Manaus,Amazonas,Manaus,False,2020-05-18T14:05:32.277Z,2020-08-01T12:03:38.649Z,2020


Para as colunas de saída talvez fosse possível calcular o valor olhando a queda para os valores de ocupação dos leitos, mas de 123170 para 124387 o número de ocupaçãoconfirmadoCli diminuiu sem valor para saída, logo as entradas que tem valor nulo para saidaSuspeitaObitos (116), saidaSuspeitaAltas (104), saidaConfirmadaObitos (12) e saidaConfirmadaAltas(13) serão removidas , além disso elas não são um número muito significativo para o volume de dados em geral.


In [168]:
print(df['saidaSuspeitaObitos'].isna().sum())
print(df['saidaSuspeitaAltas'].isna().sum())
print(df['saidaConfirmadaObitos'].isna().sum())
print(df['saidaConfirmadaAltas'].isna().sum())

116
104
12
13


In [169]:
df = df.dropna(subset=['saidaSuspeitaObitos'])
df = df.dropna(subset=['saidaSuspeitaAltas'])
df = df.dropna(subset=['saidaConfirmadaObitos'])
df = df.dropna(subset=['saidaConfirmadaAltas'])

In [170]:
print(df['saidaSuspeitaObitos'].isna().sum())
print(df['saidaSuspeitaAltas'].isna().sum())
print(df['saidaConfirmadaObitos'].isna().sum())
print(df['saidaConfirmadaAltas'].isna().sum())

0
0
0
0


## Ocupacao Covid

Para os valores de 2020 e 2021, as ocupações dos leitos eram divididas em ocupação por suspeita e ocupação quando o diagnóstico de COVID era confirmado, isso levou ao grande número de NaNs nas colunas referentes às ocupações dos leitos, para resolver isso os campos NaN de ocupacaoCovidUti e ocupacaoCovidCli somando ocupacaoSuspeito e ocupacaoConfirmado para Cli ou UTI correspondentemente

In [171]:
mask = (
    df['ano_origem'].isin(['2020', '2021']) &
    df['ocupacaoCovidUti'].isna()
)

df.loc[mask, 'ocupacaoCovidUti'] = (
    df.loc[mask, 'ocupacaoSuspeitoUti'].fillna(0) +
    df.loc[mask, 'ocupacaoConfirmadoUti'].fillna(0)
)

mask = (
    df['ano_origem'].isin(['2020', '2021']) &
    df['ocupacaoCovidCli'].isna()
)

df.loc[mask, 'ocupacaoCovidCli'] = (
    df.loc[mask, 'ocupacaoSuspeitoCli'].fillna(0) +
    df.loc[mask, 'ocupacaoConfirmadoCli'].fillna(0)
)

In [172]:
df.head()

,Unnamed: 0,_id,dataNotificacao,cnes,ocupacaoSuspeitoCli,ocupacaoSuspeitoUti,ocupacaoConfirmadoCli,ocupacaoConfirmadoUti,ocupacaoCovidUti,ocupacaoCovidCli,ocupacaoHospitalarUti,ocupacaoHospitalarCli,saidaSuspeitaObitos,saidaSuspeitaAltas,saidaConfirmadaObitos,saidaConfirmadaAltas,origem,_p_usuario,estadoNotificacao,municipioNotificacao,estado,municipio,excluido,_created_at,_updated_at,ano_origem
0,0,p5Ez41Zu6B,2020-04-14T03:00:00.000Z,2303167,0.0,0.0,0.0,0.0,0.0,0.0,NaN,NaN,0.0,0.0,0.0,0.0,aplicacao-web,_User$auVMjEVEVz,Santa Catarina,Itapema,Santa Catarina,Itapema,True,2020-04-15T13:44:30.085Z,2020-04-16T13:56:13.656Z,2020
1,1,eUOMsSt7T7,2020-04-15T03:00:00.000Z,5935377,1.0,0.0,0.0,0.0,0.0,1.0,NaN,NaN,0.0,1.0,0.0,0.0,parse-cloud,_User$WBa5nfe9P9,Rio de Janeiro,Rio de Janeiro,Rio de Janeiro,Rio de Janeiro,False,2020-04-15T13:50:47.256Z,2020-08-01T20:08:13.116Z,2020
2,2,QaG4oIBrMa,2020-04-15T14:57:16.255Z,12599,15.0,1.0,3.0,2.0,3.0,18.0,NaN,NaN,0.0,0.0,0.0,0.0,parse-cloud,_User$6oxZQHBPQ7,Rio de Janeiro,Niterói,Rio de Janeiro,Niterói,False,2020-04-15T15:20:57.033Z,2020-08-01T20:02:12.729Z,2020
3,3,ifya31F3ZF,2020-03-24T03:00:00.000Z,2562871,9.0,0.0,1.0,0.0,0.0,10.0,NaN,NaN,0.0,0.0,0.0,0.0,parse-cloud,_User$fhufIbmR9C,Ceará,Várzea Alegre,Ceará,Várzea Alegre,False,2020-04-15T16:58:32.839Z,2020-08-01T21:40:37.624Z,2020
4,4,dkurDhelx9,2020-04-01T03:00:00.000Z,2252228,26.0,0.0,1.0,0.0,0.0,27.0,NaN,NaN,0.0,0.0,0.0,0.0,parse-cloud,_User$tkAdHTNhC9,Rio Grande do Sul,Encantado,Rio Grande do Sul,Encantado,False,2020-04-15T17:02:53.877Z,2020-08-01T21:37:53.664Z,2020


In [173]:
print("Valores ausentes por coluna:")
print(df.isnull().sum())

Valores ausentes por coluna:
Unnamed: 0                     0
_id                            0
dataNotificacao                0
cnes                           0
ocupacaoSuspeitoCli       575203
ocupacaoSuspeitoUti       575203
ocupacaoConfirmadoCli     575202
ocupacaoConfirmadoUti     575202
ocupacaoCovidUti               0
ocupacaoCovidCli               0
ocupacaoHospitalarUti    1007840
ocupacaoHospitalarCli    1007840
saidaSuspeitaObitos            0
saidaSuspeitaAltas             0
saidaConfirmadaObitos          0
saidaConfirmadaAltas           0
origem                         0
_p_usuario                     0
estadoNotificacao              0
municipioNotificacao           0
estado                         1
municipio                      1
excluido                       0
_created_at                    0
_updated_at                    0
ano_origem                     0
dtype: int64


# Tipo de dados

Alterar o tipo de dataNotificao,created_at e updated_at para datetime64, assim essa coluna poderá ser usada para gerar Dia, Mês e Hora.

In [174]:
df['dataNotificacao'] = pd.to_datetime(df['dataNotificacao'])
df['dataCriado'] = pd.to_datetime(df['_created_at'])
df['dataAtualizado'] = pd.to_datetime(df['_updated_at'])

In [175]:
df['dia_notificacao'] = df['dataNotificacao'].dt.day
df['mes_notificacao'] = df['dataNotificacao'].dt.month
df['hora_notificao'] = pd.to_datetime(df['dataNotificacao']).dt.time

df['dia_criado'] = df['dataCriado'].dt.day
df['mes_criado'] = df['dataCriado'].dt.month
df['hora_criado'] = pd.to_datetime(df['dataCriado']).dt.time

df['dia_atualizado'] = df['dataAtualizado'].dt.day
df['mes_atualizado'] = df['dataAtualizado'].dt.month
df['hora_atualizado'] = pd.to_datetime(df['dataAtualizado']).dt.time



In [176]:
df.head()

,Unnamed: 0,_id,dataNotificacao,cnes,ocupacaoSuspeitoCli,ocupacaoSuspeitoUti,ocupacaoConfirmadoCli,ocupacaoConfirmadoUti,ocupacaoCovidUti,ocupacaoCovidCli,ocupacaoHospitalarUti,ocupacaoHospitalarCli,saidaSuspeitaObitos,saidaSuspeitaAltas,saidaConfirmadaObitos,saidaConfirmadaAltas,origem,_p_usuario,estadoNotificacao,municipioNotificacao,estado,municipio,excluido,_created_at,_updated_at,ano_origem,dataCriado,dataAtualizado,dia_notificacao,mes_notificacao,hora_notificao,dia_criado,mes_criado,hora_criado,dia_atualizado,mes_atualizado,hora_atualizado
0,0,p5Ez41Zu6B,2020-04-14 03:00:00+00:00,2303167,0.0,0.0,0.0,0.0,0.0,0.0,NaN,NaN,0.0,0.0,0.0,0.0,aplicacao-web,_User$auVMjEVEVz,Santa Catarina,Itapema,Santa Catarina,Itapema,True,2020-04-15T13:44:30.085Z,2020-04-16T13:56:13.656Z,2020,2020-04-15 13:44:30.085000+00:00,2020-04-16 13:56:13.656000+00:00,14,4,03:00:00,15,4,13:44:30.085000,16,4,13:56:13.656000
1,1,eUOMsSt7T7,2020-04-15 03:00:00+00:00,5935377,1.0,0.0,0.0,0.0,0.0,1.0,NaN,NaN,0.0,1.0,0.0,0.0,parse-cloud,_User$WBa5nfe9P9,Rio de Janeiro,Rio de Janeiro,Rio de Janeiro,Rio de Janeiro,False,2020-04-15T13:50:47.256Z,2020-08-01T20:08:13.116Z,2020,2020-04-15 13:50:47.256000+00:00,2020-08-01 20:08:13.116000+00:00,15,4,03:00:00,15,4,13:50:47.256000,1,8,20:08:13.116000
2,2,QaG4oIBrMa,2020-04-15 14:57:16.255000+00:00,12599,15.0,1.0,3.0,2.0,3.0,18.0,NaN,NaN,0.0,0.0,0.0,0.0,parse-cloud,_User$6oxZQHBPQ7,Rio de Janeiro,Niterói,Rio de Janeiro,Niterói,False,2020-04-15T15:20:57.033Z,2020-08-01T20:02:12.729Z,2020,2020-04-15 15:20:57.033000+00:00,2020-08-01 20:02:12.729000+00:00,15,4,14:57:16.255000,15,4,15:20:57.033000,1,8,20:02:12.729000
3,3,ifya31F3ZF,2020-03-24 03:00:00+00:00,2562871,9.0,0.0,1.0,0.0,0.0,10.0,NaN,NaN,0.0,0.0,0.0,0.0,parse-cloud,_User$fhufIbmR9C,Ceará,Várzea Alegre,Ceará,Várzea Alegre,False,2020-04-15T16:58:32.839Z,2020-08-01T21:40:37.624Z,2020,2020-04-15 16:58:32.839000+00:00,2020-08-01 21:40:37.624000+00:00,24,3,03:00:00,15,4,16:58:32.839000,1,8,21:40:37.624000
4,4,dkurDhelx9,2020-04-01 03:00:00+00:00,2252228,26.0,0.0,1.0,0.0,0.0,27.0,NaN,NaN,0.0,0.0,0.0,0.0,parse-cloud,_User$tkAdHTNhC9,Rio Grande do Sul,Encantado,Rio Grande do Sul,Encantado,False,2020-04-15T17:02:53.877Z,2020-08-01T21:37:53.664Z,2020,2020-04-15 17:02:53.877000+00:00,2020-08-01 21:37:53.664000+00:00,1,4,03:00:00,15,4,17:02:53.877000,1,8,21:37:53.664000


In [189]:
df = df.rename(columns={'dataNotificacao': 'data_notificacao'})
df = df.rename(columns={'_created_at': 'created_at'})
df = df.rename(columns={'_updated_at': 'updated_at'})

# Removendo colunas

In [178]:
df.head()

,Unnamed: 0,_id,data_notificacao,cnes,ocupacaoSuspeitoCli,ocupacaoSuspeitoUti,ocupacaoConfirmadoCli,ocupacaoConfirmadoUti,ocupacaoCovidUti,ocupacaoCovidCli,ocupacaoHospitalarUti,ocupacaoHospitalarCli,saidaSuspeitaObitos,saidaSuspeitaAltas,saidaConfirmadaObitos,saidaConfirmadaAltas,origem,_p_usuario,estadoNotificacao,municipioNotificacao,estado,municipio,excluido,_created_at,_updated_at,ano_origem,dataCriado,dataAtualizado,dia_notificacao,mes_notificacao,hora_notificao,dia_criado,mes_criado,hora_criado,dia_atualizado,mes_atualizado,hora_atualizado
0,0,p5Ez41Zu6B,2020-04-14 03:00:00+00:00,2303167,0.0,0.0,0.0,0.0,0.0,0.0,NaN,NaN,0.0,0.0,0.0,0.0,aplicacao-web,_User$auVMjEVEVz,Santa Catarina,Itapema,Santa Catarina,Itapema,True,2020-04-15T13:44:30.085Z,2020-04-16T13:56:13.656Z,2020,2020-04-15 13:44:30.085000+00:00,2020-04-16 13:56:13.656000+00:00,14,4,03:00:00,15,4,13:44:30.085000,16,4,13:56:13.656000
1,1,eUOMsSt7T7,2020-04-15 03:00:00+00:00,5935377,1.0,0.0,0.0,0.0,0.0,1.0,NaN,NaN,0.0,1.0,0.0,0.0,parse-cloud,_User$WBa5nfe9P9,Rio de Janeiro,Rio de Janeiro,Rio de Janeiro,Rio de Janeiro,False,2020-04-15T13:50:47.256Z,2020-08-01T20:08:13.116Z,2020,2020-04-15 13:50:47.256000+00:00,2020-08-01 20:08:13.116000+00:00,15,4,03:00:00,15,4,13:50:47.256000,1,8,20:08:13.116000
2,2,QaG4oIBrMa,2020-04-15 14:57:16.255000+00:00,12599,15.0,1.0,3.0,2.0,3.0,18.0,NaN,NaN,0.0,0.0,0.0,0.0,parse-cloud,_User$6oxZQHBPQ7,Rio de Janeiro,Niterói,Rio de Janeiro,Niterói,False,2020-04-15T15:20:57.033Z,2020-08-01T20:02:12.729Z,2020,2020-04-15 15:20:57.033000+00:00,2020-08-01 20:02:12.729000+00:00,15,4,14:57:16.255000,15,4,15:20:57.033000,1,8,20:02:12.729000
3,3,ifya31F3ZF,2020-03-24 03:00:00+00:00,2562871,9.0,0.0,1.0,0.0,0.0,10.0,NaN,NaN,0.0,0.0,0.0,0.0,parse-cloud,_User$fhufIbmR9C,Ceará,Várzea Alegre,Ceará,Várzea Alegre,False,2020-04-15T16:58:32.839Z,2020-08-01T21:40:37.624Z,2020,2020-04-15 16:58:32.839000+00:00,2020-08-01 21:40:37.624000+00:00,24,3,03:00:00,15,4,16:58:32.839000,1,8,21:40:37.624000
4,4,dkurDhelx9,2020-04-01 03:00:00+00:00,2252228,26.0,0.0,1.0,0.0,0.0,27.0,NaN,NaN,0.0,0.0,0.0,0.0,parse-cloud,_User$tkAdHTNhC9,Rio Grande do Sul,Encantado,Rio Grande do Sul,Encantado,False,2020-04-15T17:02:53.877Z,2020-08-01T21:37:53.664Z,2020,2020-04-15 17:02:53.877000+00:00,2020-08-01 21:37:53.664000+00:00,1,4,03:00:00,15,4,17:02:53.877000,1,8,21:37:53.664000


In [179]:
df = df.drop(columns=[
    'Unnamed: 0',
    'dataCriado',
    'dataAtualizado',
   
])


In [180]:
df = df.drop(columns=[
    'ocupacaoSuspeitoUti',
    'ocupacaoConfirmadoUti',
    'ocupacaoSuspeitoCli',
    'ocupacaoConfirmadoCli',
    'ocupacaoHospitalarCli',
    'ocupacaoHospitalarUti'
])

In [181]:
df.head()

,_id,data_notificacao,cnes,ocupacaoCovidUti,ocupacaoCovidCli,saidaSuspeitaObitos,saidaSuspeitaAltas,saidaConfirmadaObitos,saidaConfirmadaAltas,origem,_p_usuario,estadoNotificacao,municipioNotificacao,estado,municipio,excluido,_created_at,_updated_at,ano_origem,dia_notificacao,mes_notificacao,hora_notificao,dia_criado,mes_criado,hora_criado,dia_atualizado,mes_atualizado,hora_atualizado
0,p5Ez41Zu6B,2020-04-14 03:00:00+00:00,2303167,0.0,0.0,0.0,0.0,0.0,0.0,aplicacao-web,_User$auVMjEVEVz,Santa Catarina,Itapema,Santa Catarina,Itapema,True,2020-04-15T13:44:30.085Z,2020-04-16T13:56:13.656Z,2020,14,4,03:00:00,15,4,13:44:30.085000,16,4,13:56:13.656000
1,eUOMsSt7T7,2020-04-15 03:00:00+00:00,5935377,0.0,1.0,0.0,1.0,0.0,0.0,parse-cloud,_User$WBa5nfe9P9,Rio de Janeiro,Rio de Janeiro,Rio de Janeiro,Rio de Janeiro,False,2020-04-15T13:50:47.256Z,2020-08-01T20:08:13.116Z,2020,15,4,03:00:00,15,4,13:50:47.256000,1,8,20:08:13.116000
2,QaG4oIBrMa,2020-04-15 14:57:16.255000+00:00,12599,3.0,18.0,0.0,0.0,0.0,0.0,parse-cloud,_User$6oxZQHBPQ7,Rio de Janeiro,Niterói,Rio de Janeiro,Niterói,False,2020-04-15T15:20:57.033Z,2020-08-01T20:02:12.729Z,2020,15,4,14:57:16.255000,15,4,15:20:57.033000,1,8,20:02:12.729000
3,ifya31F3ZF,2020-03-24 03:00:00+00:00,2562871,0.0,10.0,0.0,0.0,0.0,0.0,parse-cloud,_User$fhufIbmR9C,Ceará,Várzea Alegre,Ceará,Várzea Alegre,False,2020-04-15T16:58:32.839Z,2020-08-01T21:40:37.624Z,2020,24,3,03:00:00,15,4,16:58:32.839000,1,8,21:40:37.624000
4,dkurDhelx9,2020-04-01 03:00:00+00:00,2252228,0.0,27.0,0.0,0.0,0.0,0.0,parse-cloud,_User$tkAdHTNhC9,Rio Grande do Sul,Encantado,Rio Grande do Sul,Encantado,False,2020-04-15T17:02:53.877Z,2020-08-01T21:37:53.664Z,2020,1,4,03:00:00,15,4,17:02:53.877000,1,8,21:37:53.664000


# Conectando

In [ ]:
%pip install sqlalchemy
%pip install psycopg2-binary

In [190]:
colunas_hospital = ['cnes','estado','municipio']

dim_hospital = df[colunas_hospital].drop_duplicates().reset_index(drop=True)
dim_hospital = dim_hospital.reset_index().rename(columns={'index': 'id_hospital_sk'})

print(f"Dimensão Hospital: {len(dim_hospital)} perfis")

colunas_notificacao = ['origem','_p_usuario','estadoNotificacao','municipioNotificacao']

dim_notificacao = df[colunas_notificacao].drop_duplicates().reset_index(drop=True)
dim_notificacao = dim_notificacao.reset_index().rename(columns={'index': 'id_notificacao_sk'})

print(f"Dimensão Notificação: {len(dim_notificacao)} perfis")

colunas_tempo = ['data_notificacao','created_at','updated_at','ano_origem','dia_notificacao','mes_notificacao','hora_notificao','dia_criado','mes_criado','hora_criado','dia_atualizado','mes_atualizado','hora_atualizado']

dim_tempo = df[colunas_tempo].drop_duplicates().reset_index(drop=True)
dim_tempo = dim_tempo.reset_index().rename(columns={'index': 'id_tempo_sk'})

print(f"Dimensão Tempo: {len(dim_tempo)} perfis")

Dimensão Hospital: 4508 perfis
Dimensão Notificação: 6558 perfis
Dimensão Tempo: 1589866 perfis


In [191]:
dim_hospital.head()

,id_hospital_sk,cnes,estado,municipio
0,0,2303167,Santa Catarina,Itapema
1,1,5935377,Rio de Janeiro,Rio de Janeiro
2,2,12599,Rio de Janeiro,Niterói
3,3,2562871,Ceará,Várzea Alegre
4,4,2252228,Rio Grande do Sul,Encantado


In [192]:
dim_tempo.head()

,id_tempo_sk,data_notificacao,created_at,updated_at,ano_origem,dia_notificacao,mes_notificacao,hora_notificao,dia_criado,mes_criado,hora_criado,dia_atualizado,mes_atualizado,hora_atualizado
0,0,2020-04-14 03:00:00+00:00,2020-04-15T13:44:30.085Z,2020-04-16T13:56:13.656Z,2020,14,4,03:00:00,15,4,13:44:30.085000,16,4,13:56:13.656000
1,1,2020-04-15 03:00:00+00:00,2020-04-15T13:50:47.256Z,2020-08-01T20:08:13.116Z,2020,15,4,03:00:00,15,4,13:50:47.256000,1,8,20:08:13.116000
2,2,2020-04-15 14:57:16.255000+00:00,2020-04-15T15:20:57.033Z,2020-08-01T20:02:12.729Z,2020,15,4,14:57:16.255000,15,4,15:20:57.033000,1,8,20:02:12.729000
3,3,2020-03-24 03:00:00+00:00,2020-04-15T16:58:32.839Z,2020-08-01T21:40:37.624Z,2020,24,3,03:00:00,15,4,16:58:32.839000,1,8,21:40:37.624000
4,4,2020-04-01 03:00:00+00:00,2020-04-15T17:02:53.877Z,2020-08-01T21:37:53.664Z,2020,1,4,03:00:00,15,4,17:02:53.877000,1,8,21:37:53.664000


In [186]:
dim_notificacao.head()

,id_notificacao_sk,origem,_p_usuario,estadoNotificacao,municipioNotificacao
0,0,aplicacao-web,_User$auVMjEVEVz,Santa Catarina,Itapema
1,1,parse-cloud,_User$WBa5nfe9P9,Rio de Janeiro,Rio de Janeiro
2,2,parse-cloud,_User$6oxZQHBPQ7,Rio de Janeiro,Niterói
3,3,parse-cloud,_User$fhufIbmR9C,Ceará,Várzea Alegre
4,4,parse-cloud,_User$tkAdHTNhC9,Rio Grande do Sul,Encantado


In [187]:
fato = df.merge(dim_hospital, on=colunas_hospital, how='left')
fato = fato.merge(dim_notificacao, on=colunas_notificacao, how='left')
fato = fato.merge(dim_tempo, on=colunas_tempo, how='left')

#Pegando apenas os IDs e os dados 
colunas_fato = [
    'id_hospital_sk', 
    'id_notificacao_sk',
    'id_tempo_sk', 
    'ocupacaoCovidUti',
    'ocupacaoCovidCli',
    'saidaSuspeitaObitos',
    'saidaSuspeitaAltas',
    'saidaConfirmadaObitos',
    'saidaConfirmadaAltas'
]

fato_covid = fato[colunas_fato]
print(f"Fato pronta: {len(fato_covid)} registros.")

Fato pronta: 1589874 registros.


In [ ]:
from sqlalchemy import create_engine

# Configurando a conexão
usuario_db = ''               
senha_db = ''                
host_db = ''                    
porta_db = ''                    
nome_banco_db = ''         

# Criando a "connection string"
DATABASE_URL = f'postgresql://{usuario_db}:{senha_db}@{host_db}:{porta_db}/{nome_banco_db}?client_encoding=utf8'
print("Tentando criar conexão com o banco...")

try:
    engine = create_engine(DATABASE_URL)
    print("Conexão bem-sucedida! Iniciando a carga do Esquema Estrela...")
    
    print("Carregando Dimensão Hospital...")
    dim_hospital.to_sql('dim_hospital', con=engine, if_exists='replace', index=False)
    
    print("Carregando Dimensão Notificacao...")
    dim_notificacao.to_sql('dim_notificacao', con=engine, if_exists='replace', index=False)
    
    print("Carregando Dimensão Tempo...")
    dim_tempo.to_sql('dim_tempo', con=engine, if_exists='replace', index=False)
    
   

    
    print("Carregando Tabela Fato...")
    fato_covid.to_sql('fato_covid', con=engine, if_exists='replace', index=False)

    print("\nCARGA CONCLUÍDA! Todas as 6 tabelas do Esquema Estrela foram criadas.")

except Exception as e:
    print(f"\n--- ERRO! ---")
    print(f"Erro ao carregar os dados: {e}")

Tentando criar conexão com o banco...
Conexão bem-sucedida! Iniciando a carga do Esquema Estrela...
Carregando Dimensão Hospital...
Carregando Dimensão Notificacao...
Carregando Dimensão Tempo...
Carregando Tabela Fato...

CARGA CONCLUÍDA! Todas as 6 tabelas do Esquema Estrela foram criadas.
